src/scipp/coords/transform_coords.py

```diff
@@ -75,17 +82,30 @@ def transform_coords(x: Union[DataArray, Dataset],
                       keep_intermediate=keep_intermediate,
                       keep_inputs=keep_inputs,
                       quiet=quiet)
-    targets = {targets} if isinstance(targets, str) else set(targets)
-    if isinstance(x, DataArray):
-        return _transform_data_array(x,
-                                     targets=targets,
-                                     graph=Graph(graph),
-                                     options=options)
+    if targets is None:
+        targets = []
+    if graph is None:
+        graph = {}
+
+    if isinstance(targets, dict):
+        if graph:
+            raise ValueError(
+                "Got multiple transformation graphs, must provide only one.")
+        graph = targets
+        targets = set(targets)
     else:
-        return _transform_dataset(x,
-                                  targets=targets,
-                                  graph=Graph(graph),
-                                  options=options)
+        targets = {targets} if isinstance(targets, str) else set(targets)
+
+    if not set(kwargs).isdisjoint(graph):
+        raise ValueError(f"Duplicate graph nodes in transformation graph {list(graph)} "
+                         f"and keyword arguments {list(kwargs)}.")
+
+    if not targets:
+        targets = set(kwargs)
```

Comment:
Initially I handled this like the graph nodes (check for duplicates above, use `targets.update(kwargs)` here), but I think the way I do it now is more consistent with the other behavior. Essentially, `kwargs` just defines `graph` "inline", no other behavior. The only other component to this change is then that when `targets` is not given we default to computing all keys of the graph (whether given via `graph` or postionally, or via `kwargs`).

Reply:
So shouldn't this be
```.py
graph.update(kwargs)
if not targets:
        targets = set(graph)
```
?

Reply:
What about
```.py
da.transform_coords(graph=graph)
```
vs
```.py
da.transform_coords(graph=graph, a=fa)
```
The latter adds to the graph but also completely changes the output.

Reply:
The first computes nothing (since only a graph is given, but no target requested), the second computes `a`. Is that what you mean by completely changes the output?

Reply:
@jl-wynen Would most of your concerns about complexity be addressed if we changed this to? 
```suggestion
      targets.update(kwargs)
```
?